Mathlib/FieldTheory/Differential/Liouville.lean

```diff
@@ -0,0 +1,207 @@
+/-
+Copyright (c) 2024 Daniel Weber. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Daniel Weber
+-/
+import Mathlib.Algebra.Algebra.Field
+import Mathlib.FieldTheory.Differential.Basic
+import Mathlib.FieldTheory.Galois.Basic
+import Mathlib.FieldTheory.IsAlgClosed.AlgebraicClosure
+
+/-!
+# Liouville's theorem
+
+A proof of Liouville's theorem. Follows
+[Rosenlicht, M. Integration in finite terms][Rosenlicht_1972].
+
+## Liouville field extension
+
+This file defines Liouville field extensions, which are differential field extensions which satisfy
+a slight generalization of Liouville's theorem. Note that this definition doesn't appear in the
+literature, and we introduce it as part of the formalization of Liouville's theorem.
+
+## Main declarations
+- `IsLiouville`: A field extension being Liouville
+- `isLiouville_of_finiteDimensional`: all finite dimensional field extensions
+  (of a field with characteristic 0) are Liouville.
+
+-/
+
+open Differential algebraMap IntermediateField Finset Polynomial
+
+variable (F : Type*) (K : Type*) [Field F] [Field K] [Differential F] [Differential K]
+variable [Algebra F K] [DifferentialAlgebra F K]
+
+/--
+We say that a differential field extension `K / F` is Liouville if, whenever an element `a ∈ F` can
+be written as `a = v + ∑ cᵢ * logDeriv uᵢ` for `v, cᵢ, uᵢ ∈ K` and `cᵢ` constant, it can also be
+written in that way with `v, cᵢ, uᵢ ∈ F`.
+-/
+class IsLiouville : Prop where
+  isLiouville (a : F) (ι : Type) [Fintype ι] (c : ι → F) (hc : ∀ x, (c x)′ = 0)
```

Comment:
Should this be `(c : ι → K)`, with `K` instead of `F`?
Or does that not matter?

Also, would you mind using different symbols in the conclusion, instead of `c'`, `u'`, `v'`. It looks a bit too much like the differential, eg in `(c x)'`.
Maybe `c_0`, `u_0`, `v_0` instead?

Reply:
In Liouville's theorem we have a shared field of constants, so it doesn't matter. I don't know if this definition is useful in other cases, and what's the appropriate range in that case

Reply:
I'll use different symbols, but note that the differential uses ′, not ' - github simply displays them the same way in code blocks